spec/integration/outputs/index_spec.rb

```diff
@@ -0,0 +1,90 @@
+require "spec/es_spec_helper"
+
+shared_examples "an indexer" do
+    let(:index) { 10.times.collect { rand(10).to_s }.join("") }
+    let(:type) { 10.times.collect { rand(10).to_s }.join("") }
+    let(:event_count) { 10000 + rand(500) }
+    let(:flush_size) { rand(200) + 1 }
+    let(:config) { "not implemented" }
+
+    it "ships events" do
+      insist { config } != "not implemented"
+
+      pipeline = LogStash::Pipeline.new(config)
+      pipeline.run
+
+      index_url = "http://#{@es_server[:host]}:#{@es_server[:port]['http']}/#{index}"
```

Comment:
I would rather provide most of the data to build this from the helpers classes and then avoid using any kind of class variables in the test if not strictly necessary.